src/beamlime/config/preset_options.py

```diff
@@ -1,39 +1,144 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
 
-import enum
-from typing import Protocol, Union
+from typing import Protocol, TypeVar, runtime_checkable
 
 from beamlime import __name__ as RESERVED_APP_NAME  # noqa F401
 
 from .tools import find_home
 
-if hasattr(enum, "StrEnum"):
-    FlagType = enum.StrEnum
-else:
-    FlagType = enum.Flag
+PresetValueType = TypeVar("PresetValueType", str, int)
 
-HOME_DIR = find_home()  # ~/.beamlime
-DEFAULT_CONFIG_PATH = HOME_DIR.joinpath("default-config.yaml")
-DEFAULT_LOG_DIR = HOME_DIR.joinpath("logs")
-DEFAULT_CUSTOM_HANDLER_DIR = HOME_DIR.joinpath("custom-handlers")
 
-DEFAULT_TIMEOUT = 60  # s
-MAX_TIMEOUT = 600  # s
-DEFAULT_WAIT_INTERVAL = 1  # s
-MIN_WAIT_INTERVAL = 1e-2  # s
+@runtime_checkable
+class PresetOptionProtocol(Protocol):
+    @property
+    @staticmethod
+    def default() -> PresetValueType:
+        ...
 
 
-class PresetOptionProtocol(Protocol):
-    @classmethod
+@runtime_checkable
+class NumericBoundPresetOptionProtocol(PresetOptionProtocol, Protocol):
+    @property
+    @staticmethod
+    def minimum() -> PresetValueType:
+        ...
+
     @property
-    def DEFAULT(cls) -> Union[str, int]:
-        pass
+    @staticmethod
+    def maximum() -> PresetValueType:
+        ...
+
+
+class SystemClass:
+    """
+    Beamlime system class.
```

Comment:
What system? What is a system class?

Reply:
Initially it was a helper class to build a pipe line and then now it's a wrapper of application instances that controls them all at once... I'll add description and think of a better class name....